src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -769,35 +759,32 @@ void GetUserConvCoeffs(EnergyPlusData &state)
                                                                  ipsc->lAlphaFieldBlanks,
                                                                  ipsc->cAlphaFieldNames,
                                                                  ipsc->cNumericFieldNames);
-        auto &inUserCurve = state.dataConvectionCoefficient->hcIntUserCurve(Loop);
+        auto &inUserCurve = state.dataConvect->hcIntUserCurve(Loop);
         inUserCurve.Name = ipsc->cAlphaArgs(1);
+
+        ErrorObjectHeader eoh{RoutineName, CurrentModuleObject, inUserCurve.Name};
         inUserCurve.refTempType = static_cast<RefTemp>(getEnumerationValue(RefTempNamesUC, ipsc->cAlphaArgs(2)));
         if (inUserCurve.refTempType == RefTemp::Invalid) {
             ShowSevereError(state,
                             format("GetUserSuppliedConvectionCoefficients: {}: Invalid Key choice Entered, for {}={}",
                                    CurrentModuleObject,
                                    ipsc->cAlphaFieldNames(2),
                                    ipsc->cAlphaArgs(2)));
+
             ErrorsFound = true;
         }
 
         if (!ipsc->lAlphaFieldBlanks(3)) {
             inUserCurve.hcFnTempDiffCurveNum = Curve::GetCurveIndex(state, ipsc->cAlphaArgs(3));
             if (inUserCurve.hcFnTempDiffCurveNum == 0) {
-                ShowSevereError(state,
-                                format("GetUserSuppliedConvectionCoefficients: {}: Invalid Name Entered, for {}={}",
-                                       CurrentModuleObject,
-                                       ipsc->cAlphaFieldNames(3),
-                                       ipsc->cAlphaArgs(3)));
+                ShowSevereItemNotFound(state, eoh, ipsc->cAlphaFieldNames(3), ipsc->cAlphaArgs(3));
                 ErrorsFound = true;
             } else { // check type
-                ErrorsFound |= Curve::CheckCurveDims(state,
-                                                     inUserCurve.hcFnTempDiffCurveNum, // Curve index
-                                                     {1},                              // Valid dimensions
-                                                     RoutineName,                      // Routine name
-                                                     CurrentModuleObject,              // Object Type
-                                                     inUserCurve.Name,                 // Object Name
-                                                     ipsc->cAlphaFieldNames(3));       // Field Name
+                auto const *curve = state.dataCurveManager->PerfCurve(inUserCurve.hcFnTempDiffCurveNum);
```

Comment:
Here is a second and much better example. I refactored this function so that the check is done inline and the function is only for printing the error message. The original function (actually it still exists because I didn't propagate this pattern to other files) took six arguments, four of them just for the purposes of printing errors (which presumably are the uncommon case). Much better to do the common simple case inline and then call a function for the uncommon case.

Reply:
I have to ask why `auto const *` is used instead of `auto const &`

Reply:
The `Curve` array stores pointers to Curve objects so that we can make them polymorphic in the future. FWIW the `Material` array is the same and other objects that can benefit from polymorphism will be the same also in the future.